src/core.jl

```diff
@@ -52,32 +71,31 @@ else
     end
 end
 
-Signal{T}(x::T, parents=(); name::String=auto_name!("input")) =
+Signal{T}(x::T, parents=(); name::String = auto_name!("input")) =
     Signal{T}(x, parents, Dict{Signal, Int}(), name)
-Signal{T}(::Type{T}, x, parents=(); name::String=auto_name!("input")) =
+Signal{T}(::Type{T}, x, parents=(); name::String = auto_name!("input")) =
     Signal{T}(x, parents, Dict{Signal, Int}(), name)
 # A signal of types
 Signal{T}(t::Type{T}; name::String = auto_name!("input")) = Signal(Type{T}, t, name=name)
 
-log_gc(n) =
+function log_gc(n)
     @async begin
         lock(io_lock)
         print(STDERR, "Signal got gc'd. Creation backtrace:")
         Base.show_backtrace(STDERR, n.bt)
         println(STDOUT)
         unlock(io_lock)
     end
+end
 
-const nodes = WeakRef[] #stores the nodes in order of creation (which is a topological order for execution of the nodes' actions)
 
-const edges = Vector{Int}[] #parents to children, useful for plotting graphs
 
-const node_count = DefaultDict{String,Int}(0) #counts of different signals for naming
 function auto_name!(name, parents...)
     parents_str = join(map(s->s.name, parents), ", ")
     isempty(parents_str) || (name *= "($parents_str)")
-    node_count[name] += 1
-    countstr = node_count[name] > 1 ? "-$(node_count[name])" : ""
+    count = get!(node_count, name, 0) + 1
```

Comment:
`node_count` is a DefaultDict so don't need the `get!` But you could just change it to a regular Dict, and leave the `get!`, if you prefer that style.

Reply:
That's what I did ;)

Reply:
Ha lol, ok

Reply:
To be honest, I'm not a hundred percent sure if that was a problem. But globals turned out pretty fragile, so I tried to use only Base types for globals. And the stack traces pointed to auto_name! and this was, with other changes, the first one working.

Reply:
Sweet